src/components/common/GlobalContext.ts

```diff
@@ -1,5 +1,6 @@
 import { createContext, Dispatch, SetStateAction, useContext } from "react";
 
+// TODO: Should we separate the loading state from the unknown state? I don't like that we're checking the `member` property for null. Checking member + admin could be bundled together, in my opinion.
```

Comment:
While we may want to leave open the possibility of rendering different things to non-members vs members, I don't see why we'd need `null` for this, as it can be achieved by letting the initial state be false, and would only be set to true after successfully logging in. Only annoying thing about this change would be some refactoring, such as 

https://github.com/acmutsa/Portal/blob/c76c4a596609949eefc2ed732b92dc85c44b35d2/src/components/Navbar.tsx#L109C15-L109C15

You mention wanting to separate the loading state from the unknown state, and to clarify, are you talking about how member and admin are null even if they've submitted their sign-on but still in a loading phase? If I'm understanding this right, I can't think of a reason why we'd need to have a loading state for either of these, but there could definitely be something I'm not thinking of.

Reply:
So the concept I was going for was eliminating the 'nullish' portion of these types, making them straight booleans. Then, checking whether we've finished loading can be done with a separate, third variable, `loading`. The `admin` and `member` properties can be `false` by default, assuming an unauthenticated state.

The reason why we're concerned with loading is in the NavBar when we're still figuring out if a user is authenticated, and changing whether to show 'Login' or 'Register'. That flash of unauthenticated content, or whatever you want to call it, is confusing.

Like, what would you think if you logged onto a website and it said 'Login' and then 0.5 seconds later it flashes to "Hello, {name}"? That's the purpose of the skeleton components (the loading indicator).

That said, the current types bother me; we're already executing two separate routes - login checks for both the member and admin sides, every single time. Even normal users do it. And while tRPC is nice and bundles them together - I think we could do with a singular login route and abstracting the loading logic to a separate variable.

Pros:
- Single route, better performance
- Get rid of nullish values
- More explicit meaning of what is being checked with 'loading'
    - We could do "loadingAuthentication" or "loadingContext" or something else, but the other variables are about the same in terms of naming & understanding. Null values have no explicit meaning.

Cons:
- Can no longer have individual loading states for admin & member authentication - in the global context. Not much of a loss, really.
- Minor refactor. Typescript should raise all the usage points, so it shouldn't be hard to find where to refactor. Search or find usages & whatever.

Reply:
We can open this up in a separate issue if you want, and reference this review as context for it.